inline.h

```diff
@@ -2586,6 +2586,33 @@ S_my_memrchr(const char * s, const char c, const STRLEN len)
 
 #endif
 
+#ifdef USE_ITHREADS
+
+/* Make Perl_ */
+PERL_STATIC_INLINE char *
+S_mortal_getenv(const char * str)
+{
+    char * ret;
+    dTHX;
+
+    PERL_ARGS_ASSERT_MORTAL_GETENV;
+
+    ENV_READ_LOCK;
+
+    ret = getenv(str);
+
+    /* ret points to static memory in getenv(), which may be overwritten at any
+     * time; use a mortal copy instead */
+    if (ret && PL_scopestack_ix != 0) {
```

Comment:
I don't know if the PL_scopestack clause is needed or not.

Reply:
If it is intended to duplicate the logic from locale.c, some or all of the comment there should move here too to help explain it.

Reply:
Right. 

 If we keep it, explanation is needed.  The reason it is in locale.c is that SAVEFREEPV doesn't work before scope stacks are set up.  I'm not sure the same restriction applies to mortalizing an SV.

Reply:
I just pushed a new version that keeps this test and explains it, based on further inspection of the code base.   I think this function should be callable anywhere anytime, and it should adequately deal with the instances where it can't create a mortalized copy.  That turns out to be only in destruction and early construction of the interpreter.  I think that in those instances, nothing else is going on that there would be a race with, so a plain getenv() is sufficient.    I could be wrong in the case of embedded perls?

However, the caller may have to copy the returned string if it needs to save it beyond the next call of this function on some platforms.  There's nothing like that in the core now, and it seems to me that the uses of this during such extremities will be rather simple.  I don't think most people are aware of the issues with getenv() that this normally works around.  One possible way forward is to assert here that we are in an area where things are working, thus forcing other uses of it to convert to plain getenv.  That way it would be more obvious to people in the know that the getenv is potentially unsafe.